radio/src/functions.cpp

```diff
@@ -392,8 +422,17 @@ void evalFunctions(const CustomFunctionData * functions, CustomFunctionsContext
 #if defined(HARDWARE_TOUCH)
           case FUNC_DISABLE_TOUCH:
             newActiveFunctions |= (1u << FUNCTION_DISABLE_TOUCH);
-        	break;
+            break;
 #endif
+#if defined(COLORLCD)
+          case FUNC_SET_SCREEN:
+            if (isRepeatDelayElapsed(functions, functionsContext, i)) {
+              TRACE("SET VIEW %d", (CFN_PARAM(cfn)));
+              int8_t screenNumber = max(0, CFN_PARAM(cfn) - 1);
+              ViewMain::instance()->setCurrentMainView((screenNumber));
```

Comment:
LVGL is not thread safe, this should be done from the UI thread using the same mechanisms as screenshots or disabling touch (for example):
- set the function bit in `newActiveFunctions`.
- check it from UI thread (look for `isFunctionActive()`).

Reply:
Ooops... This was unexpected. What is the point of having instance() function then?